src/main/java/forum/service/ThreadServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package forum.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import forum.dao.ThreadDAO;
+import forum.entity.Category;
+import forum.entity.Thread;
+import forum.rest.CategoryRestController;
+
+@Service
+public class ThreadServiceImpl implements ThreadService {
+	
+	@Autowired
+	private ThreadDAO threadDAO;
+	
+	@Autowired
+	private CategoryRestController categoryController;
+
+	@Override
+	@Transactional
+	public Thread saveThread(Thread thread) {
+		
+		Category tempCategory = categoryController.getCategory(thread.getCategory().getId());
```

Comment:
Lines 25-27. I guess they don't have effect on database. So what is the purpose of them? Or did I miss something? :D

Reply:
I'm adding the thread to the category there so that the Spring maps the correct Category entity to the Thread entity. Without doing when I make a POST request to add a thread the server returns the thread's category's title as null.

Reply:
So, to clarify here is what POST method returns if I removed lines 25-27:

```json
{
    "id": 58,
    "title": "This here is a title",
    "content": "Content of a thread that should be in music category",
    "category": {
        "id": 2,
        "title": null
    }
}
```